tests/test_canonize.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+from datetime import datetime, timezone
+
+import preql
+
+from data_diff.database import BigQuery, MySQL, Snowflake, connect_to_uri
+from data_diff.sql import Select
+from data_diff import database as db
+
+from .common import CONN_STRINGS
+
+
+DATE_TYPES = {
+    db.Postgres: ["timestamp(6) with time zone", "timestamp(6) without time zone"],
+    db.MySQL: ["datetime(6)", "timestamp(6)"],
+    db.Snowflake: ["timestamp(6)", "timestamp_tz(6)", "timestamp_ntz(6)"],
+    db.BigQuery: ["timestamp", "datetime"],
+    db.Redshift: ["timestamp(6)", "timestampz(6)"],
+    db.Oracle: ["timestamp(n) with timezone", "timestamp(n) with local time zone"],
+    db.Presto: ["timestamp", "timestamp with zone"],
+}
+
+
+class TestCanonize(unittest.TestCase):
+    i = 0
+
+    def _new_table(self, name):
+        self.i += 1
+        return f"t_{self.i}"
+
+    def test_canonize(self):
+        all_returned = {}
+
+        for db_id, conn_string in CONN_STRINGS.items():
+            print(f"Testing {db_id}")
+
+            sample_date1 = datetime(2022, 6, 3, 12, 24, 35, 69296)
+            sample_date2 = datetime(2021, 5, 2, 11, 23, 34, 58185, tzinfo=timezone.utc)
```

Comment:
I think you need a case where the timezone is different, but the date and time are the same. That shouldn't be different. I think e.g. for BigQuery that might fail.

There are probably some cases where we have to print a warning, e.g. if you cast from a timestamp with a timezone, to one that doesn't. Or from a datetime with a timezone, to a timestamp.